src/libs/actions/TaxRate.ts

```diff
@@ -514,7 +517,6 @@ function setPolicyTaxCode(policyID: string, oldTaxCode: string, newTaxCode: stri
                 key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
                 value: {
                     taxRates: {
-                        defaultExternalID: oldTaxCode === policy?.taxRates?.defaultExternalID ? newTaxCode : policy?.taxRates?.defaultExternalID,
```

Comment:
@etCoderDysto why don't we need to set `successData`?

Reply:
For offline pattern B, we should set successData - https://github.com/Expensify/App/blob/main/contributingGuides/OFFLINE_UX.md#b---optimistic-with-feedback-pattern

Reply:
@rushatgabhane BE returns updated `defaultExternalID` and `foreignTaxDefault`. That was why I removed it from sucessData.

Reply:
I am checking if removing it causes any issue.

Reply:
@rushatgabhane When we updated default tax rate, BE will update the `defaultExternalID` and `foreignTaxDefault` with the new tax code we have updated the default tax code, and merges the data to onyx. 
<img width="1704" alt="Screenshot 2024-08-27 at 10 05 27 at night" src="https://github.com/user-attachments/assets/c36f5649-8477-4cc3-9991-75a5a081edb5">

Reply:
Just to make sure I understand, you're saying that setting the successData does cause a problem?

Reply:
@rushatgabhane I can `defaultExternalID` and `foreignTaxDefault` to the successData to be safe.

Reply:
Let's do that then.